packages/pnpm/src/cmd/list.ts

```diff
@@ -33,17 +34,23 @@ export async function render (
     parseable?: boolean,
   },
   command: string,
+  isRecursive = false,
 ) {
+  const isWhy = command === 'why'
+  if (isWhy && !args.length) {
+    throw new PnpmError('MISSING_PACKAGE_NAME', `${isRecursive ? '`pnpm recursive why`' : '`pnpm why`'} requires the package name'`)
```

Comment:
I think it is fine to log always "pnpm why". We don't really know how the command was executed. It could be `pnpm -r why`